### DISCUSSION TITLE
How to disable data caching in useAsyncData or useFetch?

### DISCUSSION THREAD

**Discussion by mrtdmrmrt** [nuxt/nuxt#25402] [Questions] [answered]
I encountered the situation in the link https://github.com/nuxt/nuxt/discussions/16407, but the initialCache: false feature has been removed, what can I use instead?
When I use the "key: `temp_${Date.now()}`" feature, it requests twice on both the server and the client and it is not SEO compatible.

**Comment by xjccc** [comment]
```
useFetch('', {
  getCachedData() {
    return null
  }
})
```
![image](https://github.com/nuxt/nuxt/assets/16478776/26f40aec-2271-40c3-b343-afa968c4d91e)

**Reply by mrtdmrmrt** [reply]
I tried but it didn't work, cache continues.
When I switch between pages, new data comes, but when I refresh the page, old data comes. I want new data to arrive when the page is refreshed.
nuxt version: 3.7.4
example code:
```js
 
const {
    data: sectionsDataRES,
    pending: sectionsPendingRES,
    refresh: sectionsRefreshRES,
  } = await useFetch(() => `${baseURL}/getSeoSections`, {
    method: 'GET',
    headers: {
      Authorization: getAuthHeader(),
    },
    getCachedData() {
      return null
    }
 })
```

**Comment by TheAlexLichter** [comment]
There is no initial caching and no "getCachedData" by default. After calling `useAsyncData` once, the data is saved but will be refetched when switching anyway (unless you prerender your site, then it will be saved in a payload on build time).

Please provide a reproduction where this fails.

**Reply by mrtdmrmrt** [reply]
I don't understand what you want to say, example code:

```js
 
const {
    data: sectionsDataRES,
    pending: sectionsPendingRES,
    refresh: sectionsRefreshRES,
  } = await useFetch(() => `${baseURL}/getSeoSections`, {
    method: 'GET',
    headers: {
      Authorization: getAuthHeader(),
    },
    getCachedData() {
      return null
    }
 })
```

**Reply by TheAlexLichter** [reply]
Can you provide a [running reproduction](https://nuxt.new/s/v3) please (e.g. via stackblitz)? ☺️ 

Can be with a placeholder API or similar, no problem

**Reply by mrtdmrmrt** [reply]
Unfortunately I cannot share
There is no same problem in the test environment, it only exists in the prod environment. The prod environment works on cloudflare, could it be related to that?

**Reply by TheAlexLichter** [reply]
Well if it does not happen in staging/test but in prod then yes, it sounds like.
but without a reproduction I am not sure how anyone should debug that :D

(PS: A reproduction does not have to contain business logic or company code. using a placeholder API is sufficient)

**Reply by mrtdmrmrt** [reply]
By the way, when we release the prod environment again, new data comes. I forgot to mention this :)

**Reply by xjccc** [reply]
> By the way, when we release the prod environment again, new data comes. I forgot to mention this :)

it not your server cache？

**Reply by mrtdmrmrt** [reply]
@xjccc  As far as we know there is not.
 Does this error seem to be caused by the server rather than nuxt?

**Reply by xjccc** [reply]
I don't know that. When i deploy my nuxt with pm2, it will cache my files, every deploy i should stop my server then restart it. Because it not the newest. But your case i don't know.

**Comment by mrtdmrmrt** [comment, marked as answer]
@xjccc , @manniL Thanks for be interested. I found the problem :)
nuxt.config.ts
`
nitro: {
    compressPublicAssets: true,
    prerender: {
      failOnError: false,
      routes: ['/', 'sitemap.xml'],
    },
  }
` 
It happened when I deleted '/' from 'prerender' 👯
